src/relax/op/op_common.h

```diff
@@ -100,12 +99,26 @@ inline TensorStructInfo GetUnaryInputTensorStructInfo(const Call& call, const Bl
  * \param OpName The name of operator to register.
  * \param RequireFloatDtype A boolean indicating if the input is required to have float dtype.
  */
-#define RELAX_REGISTER_UNARY_OP_INTERFACE(OpName, RequireFloatDtype) \
-  RELAX_UNARY_OP_INTERFACE(OpName, #OpName);                         \
-  RELAX_REGISTER_UNARY_OP(#OpName, RequireFloatDtype)
+#define RELAX_REGISTER_UNARY_OP_AND_IMPL(OpName) \
+  RELAX_UNARY_OP_INTERFACE(OpName, #OpName);     \
+  RELAX_REGISTER_UNARY_OP(#OpName)
 
-template <bool require_float_dtype>
-inline StructInfo InferStructInfoUnary(const Call& call, const BlockBuilder& ctx) {
+#define RELAX_REGISTER_UNARY_ARITH_OP_AND_IMPL(OpName, RequireFloatDtype) \
+  RELAX_REGISTER_UNARY_OP_AND_IMPL(OpName).set_attr<FInferStructInfo>(    \
+      "FInferStructInfo", InferStructInfoUnaryArith<RequireFloatDtype>)
+
+#define RELAX_REGISTER_UNARY_CHECK_OP_AND_IMPL(OpName, RequireFloatDtype) \
+  RELAX_REGISTER_UNARY_OP_AND_IMPL(OpName).set_attr<FInferStructInfo>(    \
+      "FInferStructInfo", InferStructInfoUnaryCheck<RequireFloatDtype>)
+
+#define RELAX_REGISTER_UNARY_NN_OP_AND_IMPL(OpName, OpRegName, RequireFloatDtype) \
+  RELAX_REGISTER_UNARY_OP(OpRegName).set_attr<FInferStructInfo>(                  \
+      "FInferStructInfo", InferStructInfoUnaryArith<RequireFloatDtype>);          \
+  RELAX_UNARY_OP_INTERFACE(OpName, OpRegName);
```

Comment:
After your decoupling, we can now move the two for unary ops to unary.h, and the one for NN to nn.h, as these macros are no longer used across different categories.